### PR TITLE
Add summary for hapi

### PR DESCRIPTION
**Pull request by LielinJiang**
### PR types

New features
### PR changes

APIs
### Describe

Add summary for hapi

image

### CODE REVIEW

File: python/paddle/incubate/hapi/model.py [hunk 1]
```diff
@@ -1738,6 +1738,54 @@ def _run_one_epoch(self, data_loader, callbacks, mode, logs={}):
...
+
+        Returns:
+            Dict: a summary of the network.
+
+        Examples:
+            .. code-block:: python
+
+              import paddle
+              import paddle.fluid as fluid
+              import paddle.incubate.hapi as hapi
```
Comment (saxon-zh) [inline python/paddle/incubate/hapi/model.py]: 可以先Merge做测试，之后示例代码部分需要按照新的API目录结构修改一下，两个summary接口里面都包含需要修改的API目录。

**Review by saxon-zh — DISMISSED** [review]
LGTM

File: python/paddle/incubate/hapi/model.py [hunk 2]
```diff
@@ -1738,6 +1738,54 @@ def _run_one_epoch(self, data_loader, callbacks, mode, logs={}):
             return logs, outputs
         return logs
 
+    def summary(self, input_size=None, batch_size=None, dtype=None):
```
Comment (qingqing01) [inline python/paddle/incubate/hapi/model.py]: If input_size can be got from InputSpec?

File: python/paddle/incubate/hapi/utils.py [hunk 3]
```diff
@@ -178,6 +180,184 @@ def forward(self, filename):
     return state_dict
 
 
+def summary(net, input_size, batch_size=-1, dtypes=None):
+    """Prints a string summary of the network.
+
+    Args:
+        net (Layer): the network which must be a subinstance of Layer.
+        input_size (list|tuple): size of input tensor.
+        batch_size (int, optional): batch size of input tensor, Default: -1.
```
Comment (qingqing01) [inline python/paddle/incubate/hapi/utils.py]: better to use InputSpec, not input_size and batch_size to specify the input shape

File: python/paddle/incubate/hapi/utils.py [hunk 4]
```diff
@@ -178,6 +180,184 @@ def forward(self, filename):
...
+        dtypes (str, optional): if dtypes is None, 'float32' will be used, Default: None.
+
+    Returns:
+        Dict: a summary of the network.
```
Comment (qingqing01) [inline python/paddle/incubate/hapi/utils.py]: Need more detailed info for content in this dict

File: python/paddle/incubate/hapi/utils.py [hunk 5]
```diff
@@ -178,6 +180,184 @@ def forward(self, filename):
...
+    Examples:
+        .. code-block:: python
+
+            import paddle
+
+            from paddle.nn import Conv2d, Pool2D, Linear, ReLU, Sequential
+            from paddle.incubate.hapi.utils import summary
+
+
+            class LeNetDygraph(paddle.nn.Layer):
```
Comment (qingqing01) [inline python/paddle/incubate/hapi/utils.py]: LeNetDygraph  -> LeNet

File: python/paddle/incubate/hapi/utils.py [hunk 6]
```diff
@@ -178,6 +180,184 @@ def forward(self, filename):
...
+    summary_str += "Params size (MB): %0.2f" % total_params_size + "\n"
+    summary_str += "Estimated Total Size (MB): %0.2f" % total_size + "\n"
+    summary_str += "-" * table_width + "\n"
+    # return summary
+    return summary_str, {
+        'total_params': total_params,
+        'trainable_params': trainable_params
+    }
+
+
```
Comment (qingqing01) [inline python/paddle/incubate/hapi/utils.py]: need unit testing for this sumarray, you can make a small net, check the correct of  total_params  and trainable_params

File: python/paddle/incubate/hapi/utils.py [hunk 7]
```diff
@@ -178,6 +180,184 @@ def forward(self, filename):
...
+    Returns:
+        Dict: a summary of the network.
+
+    Examples:
+        .. code-block:: python
+
+            import paddle
+
+            from paddle.nn import Conv2d, Pool2D, Linear, ReLU, Sequential
```
Comment (XiaoguangHu01) [inline python/paddle/incubate/hapi/utils.py]: Pool2D -> Pool2d 跟Conv2d保持一致

File: python/paddle/incubate/hapi/utils.py [hunk 8]
```diff
@@ -178,6 +180,184 @@ def forward(self, filename):
     return state_dict
 
 
+def summary(net, input_size, batch_size=-1, dtypes=None):
```
Comment (XiaoguangHu01) [inline python/paddle/incubate/hapi/utils.py]: 建议复用InputSpec，batch_size默认值用None

File: python/paddle/incubate/hapi/model.py (hunk 2 above)
Comment (LielinJiang) [reply python/paddle/incubate/hapi/model.py]: done, thanks.

File: python/paddle/incubate/hapi/utils.py (hunk 3 above)
Comment (LielinJiang) [reply python/paddle/incubate/hapi/utils.py]: done, thanks

File: python/paddle/incubate/hapi/utils.py (hunk 4 above)
Comment (LielinJiang) [reply python/paddle/incubate/hapi/utils.py]: done, thanks

File: python/paddle/incubate/hapi/utils.py (hunk 6 above)
Comment (LielinJiang) [reply python/paddle/incubate/hapi/utils.py]: done, thanks

File: python/paddle/incubate/hapi/utils.py (hunk 5 above)
Comment (LielinJiang) [reply python/paddle/incubate/hapi/utils.py]: done, thanks

File: python/paddle/incubate/hapi/utils.py (hunk 7 above)
Comment (LielinJiang) [reply python/paddle/incubate/hapi/utils.py]: done, thanks

File: python/paddle/incubate/hapi/utils.py (hunk 8 above)
Comment (LielinJiang) [reply python/paddle/incubate/hapi/utils.py]: done, thanks

File: python/paddle/incubate/hapi/model.py (hunk 1 above)
Comment (LielinJiang) [reply python/paddle/incubate/hapi/model.py]: done, thanks

File: python/paddle/hapi/model.py [hunk 9]
```diff
@@ -1822,6 +1822,50 @@ def _run_one_epoch(self, data_loader, callbacks, mode, logs={}):
...
+            dtypes (str, optional): if dtypes is None, 'float32' will be used, Default: None.
+
+        Returns:
+            Dict: a summary of the network including total params and total trainable params.
+
+        Examples:
+            .. code-block:: python
+
+              import paddle
+              from paddle.static import InputSpec
```
Comment (saxon-zh) [inline python/paddle/hapi/model.py:1844]: 整个代码示例需要按照2.0的API进行修改一遍，去掉fluid

paddle.disable_static
paddle.Model
paddle.vision.models.LeNet

File: python/paddle/hapi/model_summary.py [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+                    x = self.features(inputs)
+
+                    if self.num_classes > 0:
+                        x = paddle.flatten(x, 1)
+                        x = self.fc(x)
+                    return x
+
+            lenet = LeNet()
+
+            params_info = paddle.hapi.summary(lenet, (1, 28, 28))
```
Comment (saxon-zh) [inline python/paddle/hapi/model_summary.py]: 隐藏hapi目录

**Review by saxon-zh — CHANGES_REQUESTED** [review]

File: python/paddle/hapi/model.py (hunk 9 above)
Comment (LielinJiang) [reply python/paddle/hapi/model.py:1844]: done, thanks

File: python/paddle/hapi/model_summary.py (hunk 10 above)
Comment (LielinJiang) [reply python/paddle/hapi/model_summary.py]: done, thanks

**Review by saxon-zh — APPROVED** [review]
LGTM

**Review by qingqing01 — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**